File: tests/go/exampleGlobal.go
package main

import (
	"fmt"
	"time"
)

var secret = "MYSUPERSECRETKEY"
var not_after = 60 // 1 minute

func main() {
	username := "user"
	password := "pass"
	token := keygen(username, password, true)
	if token != "" {
		fmt.Println("Generated token:", token)
	} else {
		fmt.Println("No token generated.")
	}
}

func login(username, password string) bool {
	// Mock login function
	return username != "" && password != ""
}

func encodeToken(username string, nbf, exp int64, secret string) string {
	// Simplified token encoding for demonstratio
	secret = secret + username
	return fmt.Sprintf("encoded_token(username=%s, nbf=%d, exp=%d, secret=%s, algorithm=HS256)", username, nbf, exp, secret)
}

func keygen(username, password string, loginRequired bool) string {
	if loginRequired {
		if !login(username, password) {
			return ""
		}
	}

	now := time.Now().Unix()
	token := encodeToken(username, now, now+int64(not_after), secret)

	return token
}

File: tests/cs/exampleGlobal.cs
using System;

namespace ExampleGlobal
{
    class Program
    {
        public static string secret = "MYSUPERSECRETKEY";
        public static int not_after = 60; // 1 minute

        static void Main(string[] args)
        {
            string username = "user";
            string password = "pass";
            string generatedToken = Keygen(username, password, true);
            if (!string.IsNullOrEmpty(generatedToken))
            {
                Console.WriteLine("Generated token: " + generatedToken);
            }
            else
            {
                Console.WriteLine("Login failed.");
            }
        }

        public static bool Login(string username, string password)
        {
            // Mock login function
            return !string.IsNullOrEmpty(username) && !string.IsNullOrEmpty(password);
        }

        public static string EncodeToken(string username, int expiry, string secret)
        {
            // Simplified token encoding for demonstration
            secret = secret ??
            return $"encoded({username},{expiry},{secret})";
        }

        public static string Keygen(string username, string password, bool loginRequired = true)
        {
            if (loginRequired)
            {
                if (!Login(username, password))
                {
                    return null;
                }
            }

            int now = 100; // Mock current time
            string token = EncodeToken(username, now + not_after, secret);

            return token;
        }
    }
}
